src/ServiceControl.Monitoring/Http/Diagrams/MonitoredEndpointMessageType.cs

```diff
@@ -1,11 +1,67 @@
 ﻿namespace ServiceControl.Monitoring.Http.Diagrams
 {
+    using System;
     using System.Collections.Generic;
+    using NServiceBus.Logging;
 
     public class MonitoredEndpointMessageType
     {
-        public string MessageType { get; set; }
+        public string TypeName { get; set; }
+
+        public string AssemblyName { get; set; }
+
+        public string AssemblyVersion { get; set; }
+
+        public string Culture { get; set; }
+
+        public string PublicKeyToken { get; set; }
 
         public Dictionary<string, MonitoredValues> Metrics { get; } = new Dictionary<string, MonitoredValues>();
+
+
+        public static MonitoredEndpointMessageType FromTypeName(string typeName)
+        {
+            //HINT: the value we get is either empty or equals EnclosedMessageTypes header value.
+            //      What it means that when non empty the value is either TypeName or AssemblyQualifiedName
+
+            if (string.IsNullOrEmpty(typeName))
+            {
+                return new MonitoredEndpointMessageType();
+            }
+
+            var typeParts = typeName.Split(new []{','}, StringSplitOptions.RemoveEmptyEntries);
+
+            if (typeParts.Length > 1)
```

Comment:
I'm scared that we omit some kind of condition in here. Have you considered using `AssemblyName` instead of a custom parser in here?

Reply:
That's a good idea. I haven't consider using `AssemblyName` for the part after first comma. Let me try that.